### PR TITLE
KNIG-67-Create-integration-tests-for-all-already-prepared-test-cases

### PR DESCRIPTION
**Pull request by moriarturo**
1. fixed add todo integration test, now it crashes 
2. added 3 non-working routine + block integration tests : save routine with todos -> save block, update amount of todos -> save block, delete routine with todos -> save block
3. added 1 working routine + block integration test : update routine name -> save block
4. fixed update routine integration test
5. fixed naming of todo test factory methods
6. changed logic of adding routine in routine service 
7. fixed logic of deleting routine with todos but it does not work for now 
8. added copying of todos and fixed copying of template routines on new block adding
9. fixed naming of fields in VOs 
10. fixed cycling in privatedb mappers and in toString of entities and VOs 
11. removed unnecessary cascade from routines to todos

### CODE REVIEW

File: src/main/java/com/knighttodo/knighttodo/domain/TodoVO.java [hunk 1]
```diff
@@ -13,7 +13,7 @@
 @AllArgsConstructor
 @NoArgsConstructor
 @Builder
-@ToString(exclude = {"routine", "block"})
```
Comment (moriarturo) [inline src/main/java/com/knighttodo/knighttodo/domain/TodoVO.java]: Why did you remove it? The idea was that we want to display entity + its children to avoid cycling in toString

File: src/main/java/com/knighttodo/knighttodo/gateway/privatedb/representation/Routine.java [hunk 2]
```diff
@@ -31,7 +32,7 @@
 @Builder
 @AllArgsConstructor
 @NoArgsConstructor
-@ToString(exclude = {"block"})
```
Comment (moriarturo) [inline src/main/java/com/knighttodo/knighttodo/gateway/privatedb/representation/Routine.java]: Same thing, why do you exclude children here, maybe it is better to remove routine displaying from todo?

File: src/main/java/com/knighttodo/knighttodo/gateway/privatedb/repository/RoutineRepository.java [hunk 3]
```diff
@@ -7,6 +7,6 @@
 
 public interface RoutineRepository extends JpaRepository<Routine, String> {
 
-    @Query("select r from Routine r where r.templateId = r.id")
+    @Query("from Routine r left join fetch r.todos where r.templateId = r.id")
```
Comment (moriarturo) [inline src/main/java/com/knighttodo/knighttodo/gateway/privatedb/repository/RoutineRepository.java]: Do you really need left join fetch here? It is only to avoid adding @Transactional in some tests, isn't it?

File: src/main/java/com/knighttodo/knighttodo/rest/RoutineResource.java [hunk 4]
```diff
@@ -78,9 +78,9 @@
...
 
     @DeleteMapping("/{routineId}")
-    public ResponseEntity<Void> deleteRoutine(@PathVariable String routineId) {
+    public ResponseEntity<Void> deleteRoutine(@PathVariable String blockId, @PathVariable String routineId) {
         log.info("Rest request to delete routine by id : {}", routineId);
-        routineService.deleteById(routineId);
+        routineService.deleteById(blockId, routineId);
         return ResponseEntity.ok().build();
     }
 }
```
Comment (moriarturo) [inline src/main/java/com/knighttodo/knighttodo/rest/RoutineResource.java]: There is misscommunication between service and resource, cause service method accepts only one parameter to deleteById

File: src/main/java/com/knighttodo/knighttodo/service/impl/BlockServiceImpl.java [hunk 5]
```diff
@@ -1,36 +1,42 @@
...
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
 import com.knighttodo.knighttodo.domain.BlockVO;
 import com.knighttodo.knighttodo.domain.RoutineVO;
 import com.knighttodo.knighttodo.domain.TodoVO;
```
Comment (moriarturo) [inline src/main/java/com/knighttodo/knighttodo/service/impl/BlockServiceImpl.java]: whyyyyy did you change an order of the imports?)

File: src/main/java/com/knighttodo/knighttodo/service/impl/BlockServiceImpl.java [hunk 6]
```diff
@@ -70,34 +76,38 @@ public void deleteById(String blockId) {
...
+        return updatedRoutineVO;
     }
 
-    private void copyTodosToRoutine(RoutineVO routine, RoutineVO routineVO) {
-        routine.setTodos(routineVO.getTodos().stream().map(this::copyTodo).collect(Collectors.toList()));
-        routine.getTodos().forEach(todoVO -> todoVO.setRoutine(routine));
+    private void copyTodosToRoutine(RoutineVO updatedRoutineVO, RoutineVO routineVO) {
+        updatedRoutineVO
+            .setTodos(routineVO.getTodos().stream().map(todo -> copyTodo(routineVO.getBlock().getId(), todo))
+                .collect(Collectors.toList()));
```
Comment (moriarturo) [inline src/main/java/com/knighttodo/knighttodo/service/impl/BlockServiceImpl.java]: I think that updatedRoutineVO and routineVO are not good names here, at least cause really updated routine is one that does not have word 'updated' in its name

File: src/main/java/com/knighttodo/knighttodo/service/impl/BlockServiceImpl.java [hunk 7]
```diff
@@ -70,34 +76,38 @@ public void deleteById(String blockId) {
...
 
-    private TodoVO copyTodo(TodoVO todoVO) {
-        TodoVO todo = new TodoVO();
-        todo.setTodoName(todoVO.getTodoName());
-        todo.setScariness(todoVO.getScariness());
-        todo.setHardness(todoVO.getHardness());
-        todo.setReady(false);
-        return todo;
+    private TodoVO copyTodo(String blockId, TodoVO todoVO) {
+        TodoVO updatedTodo = new TodoVO();
```
Comment (moriarturo) [inline src/main/java/com/knighttodo/knighttodo/service/impl/BlockServiceImpl.java]: Same thing, I think updatedTodo is not good name here, it can be something like 'copiedTodo'

File: src/main/java/com/knighttodo/knighttodo/service/impl/BlockServiceImpl.java [hunk 8]
```diff
@@ -70,34 +76,38 @@ public void deleteById(String blockId) {
...
+        updatedTodo.setTodoName(todoVO.getTodoName());
+        updatedTodo.setScariness(todoVO.getScariness());
+        updatedTodo.setHardness(todoVO.getHardness());
+        updatedTodo.setReady(false);
+        updatedTodo.setRoutine(todoVO.getRoutine());
```
Comment (moriarturo) [inline src/main/java/com/knighttodo/knighttodo/service/impl/BlockServiceImpl.java]: Idk does it work but it seems like we are setting a reference to the incorrect routine (not copied, because an old todo should have ref to an old routine)

File: src/main/java/com/knighttodo/knighttodo/gateway/privatedb/repository/RoutineRepository.java (hunk 3 above)
Comment (FedosovMax) [reply src/main/java/com/knighttodo/knighttodo/gateway/privatedb/repository/RoutineRepository.java]: It's to be able to fetch todos from Routine template and then create copies of todos in the copied routine.

File: src/main/java/com/knighttodo/knighttodo/service/impl/BlockServiceImpl.java (hunk 5 above)
Comment (FedosovMax) [reply src/main/java/com/knighttodo/knighttodo/service/impl/BlockServiceImpl.java]: sorry, forgot to move them back. You should create your own auto code style to make it work as you want ) I'll try to fix it somehow if you explain to me one more time how should it be because I don't have any issues with their order and don't really understand what is the difference between it.

File: src/main/java/com/knighttodo/knighttodo/gateway/privatedb/repository/RoutineRepository.java (hunk 3 above)
Comment (moriarturo) [reply src/main/java/com/knighttodo/knighttodo/gateway/privatedb/repository/RoutineRepository.java]: Are you sure that it does not work without join fetch?

File: src/main/java/com/knighttodo/knighttodo/service/impl/BlockServiceImpl.java (hunk 5 above)
Comment (moriarturo) [reply src/main/java/com/knighttodo/knighttodo/service/impl/BlockServiceImpl.java]: Is it not a big deal, but this variant is strange for me because they are not sorted alphabetically. I use the same editor config and Intellij always sorts it))

File: src/main/java/com/knighttodo/knighttodo/domain/TodoVO.java (hunk 1 above)
Comment (FedosovMax) [reply src/main/java/com/knighttodo/knighttodo/domain/TodoVO.java]: moved it back.

File: src/main/java/com/knighttodo/knighttodo/gateway/privatedb/representation/Routine.java (hunk 2 above)
Comment (FedosovMax) [reply src/main/java/com/knighttodo/knighttodo/gateway/privatedb/representation/Routine.java]: Removed it here.

File: src/main/java/com/knighttodo/knighttodo/gateway/privatedb/repository/RoutineRepository.java (hunk 3 above)
Comment (FedosovMax) [reply src/main/java/com/knighttodo/knighttodo/gateway/privatedb/repository/RoutineRepository.java]: removed it at all, can't even remember why I did it previously, but supposedly it could be changed to @transactional

File: src/main/java/com/knighttodo/knighttodo/rest/RoutineResource.java (hunk 4 above)
Comment (FedosovMax) [reply src/main/java/com/knighttodo/knighttodo/rest/RoutineResource.java]: fixed it. Thanks.

File: src/main/java/com/knighttodo/knighttodo/service/impl/BlockServiceImpl.java (hunk 5 above)
Comment (FedosovMax) [reply src/main/java/com/knighttodo/knighttodo/service/impl/BlockServiceImpl.java]: sorry, forgot to move them back. You should create your own auto code style to make it work as you want ) I'll try to fix it somehow if you explain to me one more time how should it be, because I don't have any issues with their order and don't really understand what is the difference between it.

File: src/main/java/com/knighttodo/knighttodo/service/impl/BlockServiceImpl.java (hunk 6 above)
Comment (FedosovMax) [reply src/main/java/com/knighttodo/knighttodo/service/impl/BlockServiceImpl.java]: Changed it to copiedRoutine and routineTemplate

File: src/main/java/com/knighttodo/knighttodo/service/impl/BlockServiceImpl.java (hunk 7 above)
Comment (FedosovMax) [reply src/main/java/com/knighttodo/knighttodo/service/impl/BlockServiceImpl.java]: yeah, thanks. It's much better.

File: src/main/java/com/knighttodo/knighttodo/service/impl/BlockServiceImpl.java (hunk 8 above)
Comment (FedosovMax) [reply src/main/java/com/knighttodo/knighttodo/service/impl/BlockServiceImpl.java]: It wasn't working right, it's nice that you understand it even without running the code)

**Review by FedosovMax — APPROVED** [review]
